src/appShell/App/Container.tsx

```diff
@@ -10,31 +11,86 @@ interface IContainerProps {
     children: React.ReactNode;
 }
 
-export default class Container extends React.Component<IContainerProps, void> {
+interface IContainerState {
+    showModal: boolean;
+    ieVersion: string;
+}
+
+export default class Container extends React.Component<IContainerProps, IContainerState> {
+
+    constructor() {
+        super();
+
+        this.state = {showModal: false, ieVersion: ''};
+        this.handleCloseClick = this.handleCloseClick.bind(this);
+    }
 
     static contextTypes = {
         router: React.PropTypes.object,
     };
 
     context: {router: any};
 
+    findIEVersion(userAgent:string, versionIndex: number, localStorage:boolean = false) {
```

Comment:
Also, it would be nice to package this method with the modal and not pollute the Container with this logic.   So, just always include the UnsupportedBrowserModal and it either renders something or not.  It manages its own state.